src/plugins/toolbar/README.md

```diff
@@ -0,0 +1,99 @@
+---
+title: Toolbar
+description: Attach a toolbar for plugins to easily register buttons on the top of a code block.
+owner: mAAdhaTTah
+body_classes: language-markup
+resources: ./demo.js { defer }
+---
+
+<section data-toolbar-order="select-code,hello-world,label">
+
+# How to use
+
+The Toolbar plugin allows for several methods to register your button, using the `Prism.plugins.toolbar.registerButton` function.
+
+The simplest method is through the HTML API. Add a `data-label` attribute to the `pre` element, and the Toolbar plugin will read the value of that attribute and append a label to the code snippet.
+
+```html { data-label="Hello World!" }
+<pre data-src="./prism-toolbar.js" data-label="Hello World!"></pre>
+```
+
+If you want to provide arbitrary HTML to the label, create a `template` element with the HTML you want in the label, and provide the `template` element's `id` to `data-label`. The Toolbar plugin will use the template's content for the button. You can also use to declare your event handlers inline:
+
+```html { data-label="my-label-button" }
+<pre data-src="./prism-toolbar.js" data-label="my-label-button"></pre>
+```
+
+```html
+<template id="my-label-button"><button onclick="console.log('This is an inline-handler');">My button</button></template>
+```
+
+## Registering buttons
+
+For more flexibility, the Toolbar exposes a JavaScript function that can be used to register new buttons or labels to the Toolbar, `Prism.plugins.toolbar.registerButton`.
+
+The function accepts a key for the button and an object with a `text` property string and an optional `onClick` function or a `url` string. The `onClick` function will be called when the button is clicked, while the `url` property will be set to the anchor tag's `href`.
+
+```js
+Prism.plugins.toolbar.registerButton("hello-world", {
+	text: "Hello World!", // required
+	onClick: function (env) {
+		// optional
+		alert(`This code snippet is written in ${env.language}.`);
+	},
+});
+```
+
+See how the above code registers the `Hello World!` button? You can use this in your plugins to register your own buttons with the toolbar.
+
+If you need more control, you can provide a function to `registerButton` that returns either a `span`, `a`, or `button` element.
+
+```js
```

Comment:
This applies to other `demo.js` files (but possibly not all): Instead of including a separate `demo.js` that basically duplicates what is here, can we have a class or something that we can add to code examples and have them get a "Run" button?

Reply:
Yes, that makes sense. Though I think there is a `demo.js` file that should have another name: it's just a script that runs some functionality on the docs page, not related to demos (IIRC, it's related to the autoloader plugin). I should revert its name to `script.js`.